### PR TITLE
ezCar V 1.0 Front-End

### PR DESCRIPTION
**Pull request by BregornOriginal**
# The final ezCar V 1.0 has been finished! 🥳 💯 🥇  
 
***We have been working on this application for the last 11 days, this application has been created on two different repositories, this one for the Front End and this for the [Back-end](https://github.com/indigodavid/ezCar_backend).***

This is Pull Request [LINK to the TO BACK-END](https://github.com/indigodavid/ezCar_backend/pull/37)

In this project, we were working on different tasks solo, and on other tasks, we were working as a group in programming. 

To assign the different tasks we have created a [Kanban board](https://github.com/indigodavid/ezCar_backend/projects/1) with the different tasks for each member and for group activities.

### Technologies

🔷 React-Redux

🔷 ES6

🔷 CSS

🔷 Webpack

🔷 Tailwind

### Tools

💠VISUAL STUDIO CODE

💠GIT

💠REDUX EXTENSION

💠REDUX LOGGER

💠LINTERS

💠Netlify

image

## ezCar has the following features

1. Frontend

     1. **HOME PAGE**
        1. Welcome React component
            ✅ *Create a Welcome page that displays a button to log in or sign up for the rest of the elements*
            ✅ *Use design to describe the App goal*

    3. **MAIN PAGE DESIGN**
        1. Create React routes
            ✅ *Create routes for all the elements in the frontend*
            ✅ *Add links to the navbar*
            ✅ *Create private routes for the user's logged*
        3. Common elements
            ✅ *Create a Navbar*
            ✅ *Create a Footer*

    4. **RESERVE DESIGN**
        1. Create a form
            ✅ *Create a React form that allows users to send a `POST` request to the API*
            ✅ *The user can select from different cards a car to reserve*
            ✅ *Include date picker*
            ✅ *The user can't select days before the actual one, and the user can't reserve days that are already reserved*
        2. Style the form
            ✅ *Use Tailwind to apply consistent styling to the reserve form*

    5. **MY RESERVATIONS DESIGN**
        1. Render on My reservation page
            ✅ *Render each reservation that can be invoked using a list from the store on Redux*
            ✅ *Render only the reservations of the user that's logged on to the app*
        2. Create a list view
            ✅ *Call the API to `GET` the reservations*

    6. **ADD CAR (ONLY FOR ADMINS)**
        1. Create a form to add car
            ✅ *Create a React form that allows users to send a Post request to the API*
            ✅ *The admin user can create a car to be reserved for other users*
        2. Style the form
            ✅ *`Create car` button sends a `POST` request to the API*
            ✅ *Use Tailwind to apply consistent styling to the reserve form*

    7. **DELETE CAR (ONLY FOR ADMINS)**
        1. Create a list for removing cars
            ✅ *Display all the items and allow to choose which ones to remove*
        2. Create a React button
            ✅ *`Delete` button sends a `DELETE` request to the API*

    8. **DETAILS PAGE**
        1. Create the car components
            ✅ *Create individual React components to display the car details*
        2. Style the components
            ✅ *Use Tailwind to style the details components*
            ✅ *Create a Carousel for the images*

    9. **DESKTOP**
        1. Create a version for screens bigger than 768px
            ✅ *Adjust the style of the existing components*

    10. **REDUX**
    11. Create the asynchronous actions
        1. Create the Store
            ✅ *Configure the store with `combineReducers` to `user`, `cars`, and `reservations`*
        2. Create Slices
           1. Create a user slice
           ✅ *Actions to dispatch `logIn`*
           ✅ *Actions to dispatch `logOut`*
           ✅ *Actions to dispatch `signUp`*
           ✅ *Actions to dispatch `tokenLogger`*
           2. Create a reservation slice
           ✅ *Actions to dispatch an `addReservation`*
           ✅ *Actions to dispatch a `getReservation`*
           3. Create a car slice
           ✅ *Actions to dispatch a `createCars`*
           ✅ *Actions to dispatch a `deleteCars`*
           ✅ *Actions to dispatch a `getCars`*
        3. Create Asyncthunk to fetch data from the [API](https://github.com/indigodavid/ezCar_backend)
           ✅ *Create CRUD actions for cars and reservations*
           ✅ *Create a request for login, logout, and signup*

[Example of the app](https://drive.google.com/file/d/1T5JMq0lMzNHESfjITeAJfZ0Zu6Xo70wh/view?usp=sharing)

> This project was created by:
>
> @indigodavid 
> @BregornOriginal 
> @darikmohammed 
> @ArnaudBand 
>

### Login user

https://user-images.githubusercontent.com/56404835/199306060-71f33349-ef6f-4116-8f21-9e1a91acc27b.mov

### Home page and reserving from the detail page

https://user-images.githubusercontent.com/56404835/199313666-7f5dc356-aa5a-4c1b-b83c-8763c2f7c318.mov

### Reserving car from reserve nav link 

https://user-images.githubusercontent.com/56404835/199314095-f287db15-42d6-4457-8191-309b035c1f8c.mov

### Admin option to add and delete a car 

https://user-images.githubusercontent.com/56404835/199319560-db31706d-2551-45a7-83ea-2bff6f12cf4d.mov

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,43 +1,56 @@
 ![](https://img.shields.io/badge/Microverse-blueviolet)
 
-# ezCar frontend
-
-> This project consumes our own API to allow users to rent a car in a React UI.
+# ezCar Frontend
 
+> This application allows users to book a car present in our database. The user can set the reservation date and final date for a selected car. If the user is an administrator they can upload a car and delete it from our database.
```
Comment (Meltrust) [inline README.md:5]: - [x] Please, add a screenshot to showcase your frontend project 💪 

_(This review comment was originally mistakenly posted on the backend PR.)_

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @BregornOriginal ,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Beautiful mobile version✔️
-  App is working ✔️ 
-  Good readme ✔️
-  Nice use of Redux Logger ✔️  

You can do it

## Suggested Changes ♻️
_Check the comments under the review._

You can use as many of my suggestions as you want.  If there is anything you would like to skip - feel free to do that.  However, I strongly recommend you take them into account as they can make your code better.


Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

File: README.md (hunk 1 above)
Comment (indigodavid) [reply README.md:5]: @Meltrust Thank you for your recommendation. We implemented the change as requested

**Review by Meltrust — APPROVED** [review]
#### Hi @BregornOriginal,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- App is beautiful✔️ 
- Great readme✔️ 
- Screenshot has been implemented✔️
- The app works well ✔️  

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

You can use as many of my suggestions as you want.  If there is anything you would like to skip - feel free to do that.  However, I strongly recommend you take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏


------

**Pull request merged**